src/cli/examples/integration-test.py

```diff
@@ -0,0 +1,555 @@
+#    checks on each of the created items for the stage.  This batch processing
+#    allows testing multiple components concurrently.
+
+import logging
+import os
+import re
+import sys
+from enum import Enum
+from typing import Dict, List, Optional, Set, Tuple
+from uuid import UUID, uuid4
+
+from onefuzztypes.enums import OS, ContainerType, TaskState, VmState
+from onefuzztypes.models import Job, Pool, Repro, Scaleset
+from onefuzztypes.primitives import Directory, File
+from pydantic import BaseModel, Field
+
+from onefuzz.api import Command, Onefuzz
+from onefuzz.backend import ContainerWrapper, wait
+from onefuzz.cli import execute_api
+
+LINUX_POOL = "linux-test"
+WINDOWS_POOL = "linux-test"
+BUILD = "0"
+
+
+class TemplateType(Enum):
+    libfuzzer = "libfuzzer"
+    afl = "afl"
+    radamsa = "radamsa"
+
+
+class Integration(BaseModel):
+    template: TemplateType
+    os: OS
+    target_exe: str
+    inputs: Optional[str]
+    use_setup: bool = Field(default=False)
+    wait_for_files: List[ContainerType]
+    check_asan_log: Optional[bool] = Field(default=False)
+
+
+TARGETS: Dict[str, Integration] = {
+    "linux-trivial-crash-afl": Integration(
+        template=TemplateType.afl,
+        os=OS.linux,
+        target_exe="fuzz.exe",
+        inputs="seeds",
+        wait_for_files=[ContainerType.unique_reports],
+    ),
+    "linux-libfuzzer": Integration(
+        template=TemplateType.libfuzzer,
+        os=OS.linux,
+        target_exe="fuzz.exe",
+        inputs="seeds",
+        wait_for_files=[ContainerType.unique_reports],
+    ),
+    "linux-trivial-crash": Integration(
+        template=TemplateType.radamsa,
+        os=OS.linux,
+        target_exe="fuzz.exe",
+        inputs="seeds",
+        wait_for_files=[ContainerType.unique_reports],
+    ),
+    "linux-trivial-crash-asan": Integration(
+        template=TemplateType.radamsa,
+        os=OS.linux,
+        target_exe="fuzz.exe",
+        inputs="seeds",
+        wait_for_files=[ContainerType.unique_reports],
+        check_asan_log=True,
+    ),
+    "windows-libfuzzer": Integration(
+        template=TemplateType.libfuzzer,
+        os=OS.windows,
+        target_exe="fuzz.exe",
+        inputs="seeds",
+        wait_for_files=[ContainerType.unique_reports],
+    ),
+    # TODO: include windows build of radamsa
+    # "windows-trivial-crash": Integration(
+    #     template=TemplateType.libfuzzer,
+    #     os=OS.windows,
+    #     target_exe="fuzz.exe",
+    #     inputs="seeds",
+    #     wait_for_files=[ContainerType.unique_reports],
+    # ),
+}
+
+
+class TestOnefuzz:
+    def __init__(
+        self,
+        onefuzz: Onefuzz,
+        logger: logging.Logger,
+        *,
+        pool_size: int,
+        os_list: List[OS],
+        targets: List[str],
+    ) -> None:
+        self.of = onefuzz
+        self.logger = logger
+        self.pools: Dict[OS, Pool] = {}
+        self.project = str(uuid4()).split("-")[0]
```

Comment:
I suggest adding a suffix like "test" to make the data that depends on this easily identifiable when debugging